Some documentation improvements (#6230)
tuist/tuist: docs/docs/cloud/binary-caching.md

2. Define dependencies with protocol/interface targets instead of implementation ones, and dependency-inject implementations from the top-most targets.
3. Split frequently-modified targets into smaller ones whose likelihood of change is lower.

The above suggestions are part of the [µFeatures architecture](/guide/scale/ufeatures-architecture), which we propose as a way to structure your projects to maximize the benefits not only of binary caching but also of Xcode's capabilities.

## Recommended setup

> We recommend having a CI job that **runs in every commit in the main branch** to warm the cache. This will ensure the cache always contains binaries for the changes in `main` so local and CI branch builds build incrementally upon them.

Reviewer: ```suggestion
We recommend having a CI job that **runs in every commit in the main branch** to warm the cache. This will ensure the cache always contains binaries for the changes in `main` so local and CI branch build incrementally upon them.
```

> [!TIP] CACHE WARMING USES BINARIES
> The `tuist cache` command also makes use of the binary cache to speed up the warming.

The following are some examples of common workflows:

### A developer starts to work on a new feature

1. They create a new branch from `main`.
2. They run `tuist generate`.

> 3. Tuist the most recent binaries from `main` and generates the project with them.

Reviewer: ```suggestion
3. Tuist pulls the most recent binaries from `main` and generates the project with them.
```

### A developer pushes changes upstream

1. The CI pipeline will run `tuist build` or `tuist test` to build or test the project.
2. The workflow will pull the most recent binaries from `main` and generate the project with them.

> 3. It'll then build or test the project incrementally.

Reviewer: ```suggestion
3. It will then build or test the project incrementally.
```
